schemas/system/Permission.schema

```diff
@@ -1,14 +1,12 @@
 System({
   Role: { category: 'Role', required: true, index: true },
   Category: { category: 'Category', required: true },
-  Catalog: { category: 'Catalog' },
-  Division: { category: 'Division' },
   Access: { domain: 'UserAccess', required: true },
 
-  NaturalKey: Index('Role', 'Category'),
+  // Catalog, Subdivision, Action
+  Target: { category: 'Identifier', required: true },
 
-  CatalogOrDivision: Validate(record => {
-    const notDivision = !record.Division;
-    record.Catalog ? notDivision : !notDivision;
-  })
+  NaturalKey: Unique('Role', 'Category', 'Target'),
```

Comment:
Shouldn't we add `Access` field to this key as well?
At the moment it means that each role can have only one type of access for `Category` and `Target`, and it may be possible that a role may need different types of access (for example insert, update and read).

Reply:
You are right, I think we need `Set on Enum` type to store multiple access rights in a single record, we can use `pgsql` array type for that, `pgsql` allow us to create array of enumerated type storing everything as a bits inside single integer.

Reply:
Yeah, using masks may be an easier solution.